### DISCUSSION TITLE
Rs485 cwt soil sensor support

### DISCUSSION THREAD

**Discussion by marjankow** [arendst/Tasmota#20820] [Q&A]
I'm using 
ESP32-WROOM-32U with RS485 to TTL module

[THCPH-S (5pin probe) manual_V1.4.pdf](https://github.com/arendst/Tasmota/files/14419220/THCPH-S.5pin.probe.manual_V1.4.pdf)

Any idea or help how to guess "ModbusSend" request to the device? 
![1709036248039](https://github.com/arendst/Tasmota/assets/145488060/1a3e6dcc-8683-482f-9603-ab1f2b7ac323)
![Screenshot from 2024-02-27 13-20-02](https://github.com/arendst/Tasmota/assets/145488060/c7fb3d6c-2a23-4020-8787-7eae8b59e3b5)

**Comment by sfromis** [comment]
You should not need to "guess", but instead read the PDF document you linked, to find out what it supports.

**Reply by marjankow** [reply]
Well was trying but any response from sensor, I wander if we get any tasmota console logs from 'RS485 to TTL' module (also nothing). Or you suggesting it's not supported?

**Reply by sfromis** [reply]
I'm only suggesting that Tasmota has support for Modbus, if you make sure to have all details of hardware, wiring and config exactly right. IIRC, that includes the need to use a RS485 adapter able to automatically manage direction, which is not all of those.

**Reply by Noschvie** [reply]
Example of a supported one:
https://www.aliexpress.com/item/1005001621746811.html

**Comment by mohammednidal1988** [comment]
Hi, Im trying to connect the same exact one but still no luck, did you get it working?

**Reply by marjankow** [reply]
Not succeeded. My sensor is working with the screen kit. But with rs486 to ttl and esp32 not. I was wondering how to see if rs486 to ttl integrated with my esp32 board (without sensor connected) works can we retrieve any logs.

**Reply by mohammednidal1988** [reply]
Hi Again, 
It's been along time but I just came back to the issue , 
I got it running today by using an esp32 board which has integrated rs485 in it , I had no luck with tasmota but did it with esp32 home where I defined the baud and the slave address which is iny case 2 ,
Then asked each register sapertly , I used the addresses included in the config tool documentation 

I think the key is the ground connection , I reached this after many tries on different boards but now with shared ground I believe it works fine , I have it connected to HA

![Screenshot_2026-01-27-21-35-09-666_io homeassistant companion android](https://github.com/user-attachments/assets/500d5282-0309-47df-956e-624666a5734e)
![Screenshot_2026-01-27-21-35-27-186_io homeassistant companion android](https://github.com/user-attachments/assets/a41b2bcf-f3eb-4e84-b11e-a43c7e869d39)
![IMG-20260127-WA0003](https://github.com/user-attachments/assets/f06ebce6-85d5-4bef-b97d-853a2379ef6d)
